### PR TITLE
Fix project dir argument when running debug (#1733)

### LINKED ISSUE #1733: Project dir argument not used when running debug
### Describe the bug
When running `dbt debug --project-dir=<some directory>` in another directory to that of your dbt project, the `dbt_project.yml` is not used.

### Steps To Reproduce
With an existing dbt project:

```bash
$ cd /tmp
$ dbt debug --project-dir=<path/to/project> --profiles-dir=$HOME/.dbt/
```

we get the following output:

```
Using profiles.yml file at /app/src/pipeline/profiles.yml

Configuration:
  profiles.yml file [OK found and valid]
  dbt_project.yml file [ERROR not found]
  profile: some-profile-name [OK found]
  target: production [OK found]
```

### Expected behavior
Would expect the output to be the following:

```
Using profiles.yml file at /app/src/pipeline/profiles.yml

Configuration:
  profiles.yml file [OK found and valid]
  dbt_project.yml file [OK found and valid]
  profile: some-profile-name [OK found]
  target: production [OK found]
```

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [x] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
installed version: 0.14.0
   latest version: 0.14.1
```

Also tried with version 0.14.1. and obtained the same behavior.

**The operating system you're using:**
Ubuntu 19.04

**The output of `python --version`:**
Python 3.7.4

### Additional context
After a brief debugging session, we found the `get_nearest_project_dir()` function in `dbt/core/dbt/task/base.py`. This function is called when running `dbt compile ` and `dbt run`, but not `dbt debug`.

Inspecting the parsed arguments shows the `project_dir` attribute is set correctly.

When running `dbt debug` in the actual project directory, dbt is able to pick up `dbt_project.yml` correctly.

### PR DESCRIPTION
**Pull request by franloza**
Fixes #1733

I did not make the class `DebugTask` inherited from `RequiresProjectTask` because it raised a `dbt.exceptions.RuntimeException` in `get_nearest_project_dir()` function if `dbt_project.yml` was not found. Instead, I just used the argument `project_dir` from arguments as first choice and set current directory as fall-back value.

The included test checks that, even if a `dbt_project.yml` is in the current directory, the `--project-dir` argument is used, leading to a "not found" error.

### CODE REVIEW

**Comment by drewbanin** [comment]
Thanks for making this PR @franloza!

I gave this a quick spin locally and I noticed that the code change here doesn't have the effected specified in the PR description. Specifically, I saw that when running `dbt debug --project-dir path/to/project`, dbt still looked for the `dbt_project.yml` file in the `cwd`.

I think the relevant method codepath might be around here: https://github.com/fishtown-analytics/dbt/blob/cc491904bc7ed5c27a395e3a1936e57e4221354a/core/dbt/config/project.py#L391-L393

This method is called via `_load_project` in the `DebugTask`: https://github.com/fishtown-analytics/dbt/blob/cc491904bc7ed5c27a395e3a1936e57e4221354a/core/dbt/task/debug.py#L122-L133

I think we'll want to update the `_load_project` method to construct a Project object using `args.project_dir`. Let me know if you have any questions or further thoughts here. I'll kick off the tests regardless so we can see how they fare :)

(also: @cla-bot check)

**Comment by franloza** [comment]
Yes, you were completely right @drewbanin. As you mentioned, the added test only checked that we were looking outside current directory and returned the error in the first if statement:
```
if not os.path.exists(self.project_path):
    self.project_fail_details = FILE_NOT_FOUND
    return red('ERROR not found')
```
However, we were not covering the case of specifing an existent `project.yml` file outside current directory. I have added a test to cover this case and updated `_load_project` method to use `Project.from_project_root` instead of `Project.from_current_directory`. This way, we are using always `--project-dir` as first choice and current directory as fall-back value.

**Comment by drewbanin** [comment]
Thanks for the update @franloza! I kicked off the tests here and I'm adding @beckjake to review. This LGTM when the tests pass :)

File: core/dbt/task/debug.py [hunk 1]
```diff
@@ -125,7 +126,7 @@ def _load_project(self):
             return red('ERROR not found')
 
         try:
-            self.project = Project.from_current_directory(self.cli_vars)
+            self.project = Project.from_project_root(self.project_dir, self.cli_vars)
```
Comment (drewbanin) [inline core/dbt/task/debug.py]: Looks like this line is too long per our styleguide:
```
core/dbt/task/debug.py:129:80: E501 line too long (85 > 79 characters)
```

via: https://circleci.com/gh/fishtown-analytics/dbt/19769?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Can you split this onto two lines?

File: test/integration/049_dbt_debug_test/test_debug.py [hunk 2]
```diff
@@ -103,3 +104,27 @@ def test_postgres_badproject(self):
                 self.assertIn('ERROR invalid', line)
             elif line.strip().startswith('profiles.yml file'):
                 self.assertNotIn('ERROR invalid', line)
+
+    @use_profile('postgres')
+    def test_postgres_not_found_project_dir(self):
+        self.use_default_project()
```
Comment (beckjake) [inline test/integration/049_dbt_debug_test/test_debug.py]: You shouldn't need to call `use_default_project` unless you do things with the project/profile as part of the test - `setUp` will call it (some old tests still have it, so I understand where this came from!)

File: test/integration/049_dbt_debug_test/test_debug.py [hunk 3]
```diff
@@ -103,3 +104,27 @@ def test_postgres_badproject(self):
...
+            'invalid-key': 'not a valid key in this project'
+        }
+        os.makedirs('custom', exist_ok=True)
+        with open("custom/dbt_project.yml", 'w') as f:
+            yaml.safe_dump(project_config, f, default_flow_style=True)
+        self.run_dbt(['debug', '--project-dir', 'custom'])
+        splitout = self.capsys.readouterr().out.split('\n')
+        for line in splitout:
+            if line.strip().startswith('dbt_project.yml file'):
+                self.assertIn('ERROR invalid', line)
```
Comment (beckjake) [inline test/integration/049_dbt_debug_test/test_debug.py]: Can you please add a newline at the end of this file? looking at stuff in the terminal can get annoying without trailing newlines.

File: test/integration/049_dbt_debug_test/test_debug.py (hunk 2 above)
Comment (franloza) [reply test/integration/049_dbt_debug_test/test_debug.py]: Yes, you're right @beckjake . I did not realize that it is called in `setUp` method. I will take it into account for the next time, thanks!

**Comment by franloza** [comment]
I have fixed style code style violations and removed `self.use_default_project()` from one test. Really happy to contribute, I think you all are doing a great job with tool :+1:

**Pull request merged**